File: go/003/main.go
/*
 * ProjectEuler 003
 * Go / main.go
 * copyright (c) 2015 Susisu
 *
 * Go 1.4.1
 * `go build -o bin/main main.go`
 */


package main

import "fmt"

func main() {
    n := 600851475143
    p := 2
    for n > 1 {
        if n % p == 0 {
            n /= p
        } else {
            p++
        }
    }
    fmt.Printf("%d\n", p)
}

File: java/003/Main.java
/*
 * ProjectEuler 003
 * Java / Main.java
 * copyright (c) 2015 Susisu
 *
 * Java SE 1.8.0_25
 * `javac Main.java -d bin`
 */

public class Main {
    public static void main(String[] args) {
        long n = 600851475143L;
        long p = 2;
        while (n > 1) {
            if (n % p == 0) {
                n /= p;
            }
            else {
                p++;
            }
        }
        System.out.println(p);
    }
}
